scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -67,9 +67,17 @@
 (defn prepare-cook-expected-state-dict-for-logging
   ".toString on a pod is incredibly large. Make a version thats been elided."
   [cook-expected-state-dict]
-  (if (:launch-pod cook-expected-state-dict)
-    (assoc cook-expected-state-dict :launch-pod [:elided-for-brevity])
-    cook-expected-state-dict))
+  (let [stripped-cook-expected-state-dict
+        (try
+          (-> cook-expected-state-dict
+              (dissoc :waiting-metric-timer)
+              (dissoc :running-metric-timer))
+          (catch Throwable t
+            (log/error t "Error preparing cook expected state for logging:" cook-expected-state-dict)
+            cook-expected-state-dict))]
+    (if (:launch-pod stripped-cook-expected-state-dict)
+      (assoc stripped-cook-expected-state-dict :launch-pod [:elided-for-brevity])
```

Comment:
Should we just `dissoc` this one like we're `dissoc`ing `:waiting-metric-timer` and `:running-metric-timer`?

Reply:
There's a spot in the state machine where we use the existence or non-existence of launch-pod to decide on which action we take. Having launch pod there, but elided, lets us follow the state machine. dissoc would prevent that.

Reply:
Isn't this function only used for logging? Hurrah for immutable data structures!

Reply:
It is only used for logging. I want the log to indicate whether we did or did not have a :launch-pod contained in the dictionary. Your change precludes that.

Reply:
Ah, makes sense, thanks